partiql-common/src/metadata.rs

```diff
@@ -0,0 +1,366 @@
+use indexmap::map::Entry;
+use indexmap::IndexMap;
+use rust_decimal::Decimal;
+use std::borrow::Borrow;
+use std::fmt::Result;
+use std::fmt::{Display, Formatter};
+use std::hash::Hash;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct PartiqlMetadata<T>
+where
+    T: Eq + Clone + Hash + Borrow<str>,
+{
+    inner: IndexMap<T, PartiqlMetaValue<T>>,
+}
+
+impl<T> PartiqlMetadata<T>
```

Comment:
May want to #[allow(dead_code)] on this, since the variants may not all be used in places that partiql-lang-rust can see.

Reply:
done